src/api/spec/models/package_spec.rb

```diff
@@ -356,4 +356,78 @@
     it { expect(package_with_file.public_source_path('icon')).to eq('/public/source/home:tom/package_with_files/icon') }
     it { expect(package_with_file.public_source_path('icon', { format: :html})).to eq('/public/source/home:tom/package_with_files/icon?format=html') }
   end
+
+  describe '#what_depends_on' do
+    let(:repository) { 'openSUSE_Leap_42.1'}
+    let(:architecture) { 'x86_64' }
+    let(:url) { "#{CONFIG['source_url']}/build/#{package.project}/#{repository}/#{architecture}/_builddepinfo?package=#{package.name}&view=revpkgnames" }
+
+    it 'builds backend path correct' do
+      package.what_depends_on(repository, architecture)
+      expect(a_request(:get, url)).to have_been_made.once
+    end
+
+    context 'with no build dependencies' do
+      let(:no_dependency) { "<builddepinfo />" }
+
+      before do
+        stub_request(:get, url).and_return(body: no_dependency)
+        @result = package.what_depends_on(repository, architecture)
```

Comment:
Does the team have a preference to use instances variables or let blocks? It seems like let is more commonly used in the codebase and its the preferred way in the [rspec style guidlines](http://betterspecs.org/#let).